### ISSUE TITLE
netplugin getting started example doesnt work

### ISSUE THREAD

**Issue by thesrinivas** [contiv/netplugin#213]
The failure, I see is pasted below - (cant find contiv-net network).

vagrant@netplugin-node1:~$ netctl net create contiv-net --subnet=20.1.1.0/24 --gateway=20.1.1.254 --pkt-tag=200
vagrant@netplugin-node1:~$ docker run -itd --name=web --net=contiv-net ubuntu /bin/bash
Unable to find image 'ubuntu:latest' locally
latest: Pulling from library/ubuntu

0bf056161913: Pull complete
1796d1c62d0c: Pull complete
e24428725dd6: Pull complete
89d5d8e8bafb: Pull complete
Digest: sha256:a2b67b6107aa640044c25a03b9e06e2a2d48c95be6ac17fb1a387e75eebafd7c
Status: Downloaded newer image for ubuntu:latest
8fbce56aeca753069d2b3d9960bf1a15404bcee2f43a5265a141332ceada391a
Error response from daemon: Cannot start container 8fbce56aeca753069d2b3d9960bf1a15404bcee2f43a5265a141332ceada391a: network contiv-net not found

vagrant@netplugin-node1:~$ netctl net list
Tenant   Network     Encap type  Packet tag  Subnet       Gateway
------   -------     ----------  ----------  -------      ------
default  contiv-net  vxlan       200         20.1.1.0/24  20.1.1.254

vagrant@netplugin-node1:~$ netctl net delete contiv-net
INFO[0000] Deleting network default:contiv-net
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x472689]

goroutine 1 [running]:
github.com/contiv/netplugin/netctl.writeBody(0x0, 0xc82008ac60)
	/opt/gopath/src/github.com/contiv/netplugin/netctl/http.go:41 +0x119
github.com/contiv/netplugin/netctl.deleteURL(0xc82008ac60, 0xc820064dc0, 0x36)
	/opt/gopath/src/github.com/contiv/netplugin/netctl/http.go:72 +0x115
github.com/contiv/netplugin/netctl.deleteNetwork(0xc82008ac60)
	/opt/gopath/src/github.com/contiv/netplugin/netctl/netctl.go:211 +0x4b5
github.com/codegangsta/cli.Command.Run(0x7bc9d8, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x7f4a00, 0x10, 0x0, ...)
	/opt/gopath/src/github.com/contiv/netplugin/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:131 +0x1055
github.com/codegangsta/cli.(*App).RunAsSubcommand(0xc82008aa20, 0xc82008a900, 0x0, 0x0)
	/opt/gopath/src/github.com/contiv/netplugin/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:280 +0x1168
github.com/codegangsta/cli.Command.startApp(0x7bd8d8, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x802fa0, 0x1a, 0x0, ...)
	/opt/gopath/src/github.com/contiv/netplugin/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:209 +0x7cc
github.com/codegangsta/cli.Command.Run(0x7bd8d8, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x802fa0, 0x1a, 0x0, ...)
	/opt/gopath/src/github.com/contiv/netplugin/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:59 +0x15f1
github.com/codegangsta/cli.(*App).Run(0xc82008a7e0, 0xc8200640c0, 0x4, 0x4, 0x0, 0x0)
	/opt/gopath/src/github.com/contiv/netplugin/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:173 +0xfb4
main.main()
	/opt/gopath/src/github.com/contiv/netplugin/netctl/netctl/main.go:15 +0xb4

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1696 +0x1

**Comment by erikh** [comment]
#209 has already been created to track this. I believe a patch is being worked on now.

**Comment by jainvipin** [comment]
@thesrinivas - deletion of a network while endpoints attached to is shouldn't be allowed, (#209) tracks the issue and the fix is in works.

The main issue you saw with this is really that container failed to attach to the container and was unable to find 'contiv-net'. Would it be possible to share the netmaster and netplugin logs, which can be found in /tmp/netmaster.log and /tmp/netplugin.log
or if you `grep -i error` in those files it would help.

I am not sure about the trail of events that you went through to get into this state. When trying this out afresh in my setup again just now, things worked for me.

**Comment by thesrinivas** [comment]
grep is getting confused on these log files. I've attached the files.

vagrant@netplugin-node1:~$ grep -i error /tmp/netmaster.log
Binary file /tmp/netmaster.log matches

[contiv-logs.tar.gz](https://github.com/contiv/netplugin/files/69999/contiv-logs.tar.gz)

**Comment by jainvipin** [comment]
From the  the logs, netmaster and netplugins both spew a lot of errors including when netmaster starts
> time="2015-12-20T21:18:05Z" level=fatal msg="listen error:listen tcp :9001: bind: address already in use" 

I am curious if all you did was the following?
- make demo
- make ssh (or vagrant ssh netplugin-node1)
- and run netctl commands

It seems like netmaster was restarted, was already running. A set of steps that led you to this might be more than above, which can be easily verified if you do 'make stop' from outside the VM and repeat just the above three steps.

**Comment by thesrinivas** [comment]
Yes; the steps I tried were the following (in this order):

  506  make demo
  507  vagrant ssh netplugin-node1

    1  netctl net create contiv-net --subnet=20.1.1.0/24 --gateway=20.1.1.254 --pkt-tag=200
    2  docker run -itd --name=web --net=contiv-net ubuntu /bin/bash
    3  docker run -itd --name=db --net=contiv-net ubuntu /bin/bash

The commands (2) and (3) ran failed saying "contiv-net" not found.
I then tried to delete the contiv-net (and try to redo all the steps) which failed (you probably saw those logs).

I'll try and repeat the steps as you suggested using "make stop".
